python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -1135,6 +1135,91 @@ def plot_gnss_cn0(self):
 
         self._add_figure(name='gnss_cn0', figure=figure, title='GNSS C/N0 vs. Time')
 
+    def plot_gnss_azimuth_elevation(self):
+        """!
+        @brief Plot GNSS azimuth/elevation angles.
+        """
+        result = self.reader.read(message_types=GNSSSatelliteMessage, **self.params)
+        data = result[GNSSSatelliteMessage.MESSAGE_TYPE]
+
+        if len(data.p1_time) == 0:
+            self.logger.info('No satellite data available. Skipping azimuth/elevation plot.')
+            return
+
+        # Set up the figure.
+        figure = make_subplots(
+            rows=4, cols=1,  print_grid=False, shared_xaxes=True,
+            subplot_titles=["Azimuth Angle",
+                            "Elevation Angle", None, None],
+            specs=[[{'rowspan': 2}],
+                   [None],
+                   [{'rowspan': 2}],
+                   [None]])
+        figure['layout'].update(showlegend=True, modebar_add=['v1hovermode'])
+        figure['layout']['xaxis1'].update(title=self.p1_time_label, showticklabels=True)
+        figure['layout']['xaxis2'].update(title=self.p1_time_label, showticklabels=True)
+        figure['layout']['yaxis1'].update(title="Degrees")
+        figure['layout']['yaxis2'].update(title="Degrees")
+
+        # Assign colors to each satellite.
+        data_by_sv = GNSSSatelliteMessage.group_by_sv(data)
+        svs_by_system = defaultdict(set)
+        indices_by_system = defaultdict(list)
+        svs = sorted(list(data_by_sv.keys()))
+        color_by_sv = self._assign_colors(svs)
+
+        for i, sv in enumerate(svs[::-1]):
```

Comment:
I don't think you want to iterate backwards here. I'm guessing this was taken from the fix status plot? There we intentionally iterate in reverse order so GPS is at the top of the plot (_highest_ Y axis value), but doing it here would likely put GPS at the bottom of the legend so it doesn't match other plots.

Also doesn't seem like you use `i` at all here so I don't think you need the `enumerate()` call?